pylint_odoo/checkers/modules_odoo.py

```diff
@@ -853,16 +853,31 @@ def _check_dangerous_view_replace_wo_priority(self):
         """
         self.msg_args = []
         xml_files = self.filter_files_ext('xml')
-        for xml_file in xml_files:
-            views = self.get_xml_records(
-                os.path.join(self.module_path, xml_file), model='ir.ui.view')
+        for xml_file in self._skip_files_ext('.xml', xml_files):
+            xml_file_path = os.path.join(self.module_path, xml_file)
+            views = self.get_xml_records(xml_file_path, model='ir.ui.view')
             for view in views:
                 priority = self._get_priority(view)
                 is_replaced_field = self._is_replaced_field(view)
                 if is_replaced_field and priority < self.config.min_priority:
                     self.msg_args.append((
                         "%s:%s" % (xml_file, view.sourceline), priority,
                         self.config.min_priority))
+
+            # view template
+            xml_doc = self.parse_xml(xml_file_path)
+            for template in xml_doc.xpath("/odoo//template|/openerp//template"):
```

Comment:
how about the `data` node?

Reply:
The double `//` considers `data` node

The test_repo has the `data` node examples and it is detected

Check the following example:

 - https://github.com/OCA/pylint-odoo/blob/1ee7a910bdd51627c067a88da44178853cd1ca9c/pylint_odoo/test_repo/broken_module/model_view_odoo2.xml#L3


And check the following output:
 - https://app.travis-ci.com/github/OCA/pylint-odoo/jobs/536551020#L408

`pylint_odoo/test_repo/broken_module/model_view_odoo2.xml:17: [W7902(duplicate-xml-record-id), ]  Duplicate xml record id "data/view_model_form2_noupdate_0" in model_view2.xml:5`

It based on the following similar xpath:
 - https://github.com/OCA/pylint-odoo/blob/1ee7a910bdd51627c067a88da44178853cd1ca9c/pylint_odoo/misc.py#L536

Check the following documentation about xpath:
  - https://dev.to/masihurmaruf/locator-strategy-xpath-54p7

```txt
A single slash ‘/’ anywhere in XPath signifies to look for the element immediately inside its parent element.
A double slash ‘//’ signifies to look for any child or any grand-child (descendant) element inside the parent element.
```

Reply:
I believe the root can be a `<data>` element instead of `<odoo>`. Maybe this is what Holger was referring to.

Reply:
I think root can't be <data> since that not all xml have <data>

Check the following example
https://github.com/odoo/odoo/blob/14.0/addons/account/views/account.xml

Notice the root node is <odoo>

@hbrunn meant that we should consider odoo and data but it is already considered

Is not?

Reply:
yes that's what I meant and it's indeed covered by the double slash I didn't consider